services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/TerminalPowBlockMonitorService.java

```diff
@@ -0,0 +1,182 @@
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.config.SpecConfigMerge;
+import tech.pegasys.teku.spec.datastructures.execution.PowBlock;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.executionengine.ExecutionEngineChannel;
+import tech.pegasys.teku.spec.logic.versions.merge.helpers.MiscHelpersMerge;
+import tech.pegasys.teku.storage.client.RecentChainData;
+import tech.pegasys.teku.util.config.Constants;
+
+public class TerminalPowBlockMonitorService {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final ExecutionEngineChannel executionEngine;
+  private final AsyncRunner asyncRunner;
+  private final TerminalPowBlockMonitorChannel terminalPowBlockMonitorChannel;
+  private final AtomicBoolean stopped = new AtomicBoolean(false);
+  private final Spec spec;
+  private final RecentChainData recentChainData;
+
+  private final AtomicBoolean isMerge = new AtomicBoolean(false);
+
+  private MiscHelpersMerge miscHelpers;
+  private SpecConfigMerge specConfigMerge;
+
+  private Optional<Bytes32> maybeBlockHashTracking = Optional.empty();
+
+  public TerminalPowBlockMonitorService(
+      ExecutionEngineChannel executionEngine,
+      TerminalPowBlockMonitorChannel terminalPowBlockMonitorChannel,
+      Spec spec,
+      RecentChainData recentChainData,
+      AsyncRunner asyncRunner) {
+    this.executionEngine = executionEngine;
+    this.asyncRunner = asyncRunner;
+    this.terminalPowBlockMonitorChannel = terminalPowBlockMonitorChannel;
+    this.spec = spec;
+    this.recentChainData = recentChainData;
+  }
+
+  protected void start() {
+    monitor();
+  }
+
+  protected void stop() {
+    stopped.set(true);
+  }
+
+  private void monitor() {
+    if (stopped.get()) {
+      return;
+    }
+
+    doMonitor();
+
+    asyncRunner
+        .runAfterDelay(this::monitor, Constants.TTD_MONITOR_SERVICE_POLL_INTERVAL)
+        .reportExceptions();
+  }
+
+  private void doMonitor() {
+    if (!isMerge.get()) {
+      initMergeState();
+      if (!isMerge.get()) {
+        return;
+      }
+    }
+
+    Optional<BeaconState> beaconState = recentChainData.getBestState();
+    if (beaconState.isEmpty()) {
+      return;
+    }
+
+    if (miscHelpers.isMergeComplete(beaconState.get())) {
+      stop();
+      return;
+    }
+
+    maybeBlockHashTracking.ifPresentOrElse(
+        this::checkTerminalBlockByBlockHash, this::checkTerminalBlockByTTD);
+  }
+
+  private void initMergeState() {
+    Optional<UInt64> maybeSlot = recentChainData.getCurrentSlot();
+    if (maybeSlot.isEmpty()) {
+      return;
+    }
+
+    Optional<SpecConfigMerge> maybeMergeSpecConfig =
+        spec.getSpecConfig(spec.computeEpochAtSlot(maybeSlot.get())).toVersionMerge();
+    if (maybeMergeSpecConfig.isEmpty()) {
+      return;
+    }
+
+    specConfigMerge = maybeMergeSpecConfig.get();
+    miscHelpers = ((MiscHelpersMerge) spec.atSlot(maybeSlot.get()).miscHelpers());
```

Comment:
Rather than having to case to `MiscHelpersMerge` I think it's worth adding the `isMergeComplete` to all versions of it - phase0 and Altair would just always return false.  Doesn't work for all methods but in this case it makes life a lot simpler since it has a very clear definition for pre-merge forks.